### PR TITLE
test: Migrated `test/integration/transaction` tests to `node:test`

### LINKED ISSUE #2760: Migrate transaction integration tests to `node:test`
### Description
Migrate tests it `test/integration/transaction` to `node:test`

### PR DESCRIPTION
**Pull request by bizob2828**
## Related Issues
Closes #2760

### CODE REVIEW

File: test/integration/transaction/tracer.test.js [hunk 1]
```diff
@@ -0,0 +1,877 @@
...
+      plan.equal(tracer.getSegment(), expected, 'should have expected segment')
+      throw error
+    }, segment)
+  }
+  await plan.completed
+})
+
+test('bind + capture error', async function testThrows(t) {
+  const { agent, tracer } = t.nr
+  const listeners = process.listeners('uncaughtException')
```
Comment (bizob2828) [inline test/integration/transaction/tracer.test.js:125]: I'm going to look into using the helper that I think @jsumners-nr wrote for this

Comment (bizob2828) [reply test/integration/transaction/tracer.test.js:125]: looks like i can't use it because the handler needs context from a closure above it

Comment (jsumners-nr) [reply test/integration/transaction/tracer.test.js:125]: You would probably use https://github.com/newrelic/node-newrelic/blob/36bc0e7f97f1bbde522031942ec0cc7c1d4c1349/test/lib/temp-override-uncaught.js#L32

Your passed in handler would still retain the closure.

Comment (bizob2828) [reply test/integration/transaction/tracer.test.js:125]: I can just remove the listeners with the `temp-remove-listeners`

**Review by jsumners-nr — APPROVED** [review]

**Pull request merged**